drgn_tools/task.py

```diff
@@ -161,3 +169,93 @@ def task_cpu(task: drgn.Object) -> int:
     if has_member(task, "cpu"):
         return task.cpu.value_()
     return task_thread_info(task).cpu.value_()
+
+
+def translate_nanoseconds(task: Object) -> str:
+    """
+    :returns: conversion of nanoseconds to [dd hh:mm:ss.ms] format
+    """
+    nanosecs = task_lastrun2now(task)
+    secs = nanosecs_to_secs(nanosecs)
+    time = str(datetime.timedelta(seconds=secs))
+    return time
+
+
+def get_pid(task: Object) -> int:
+    """
+    :returns: PID of the task
+    """
+    return task.pid.value_()
+
+
+def get_ps_state(task: Object) -> str:
+    """
+    :returns: state of the task
+     D    uninterruptible sleep
+     I    Idle kernel thread
+     R    running or runnable
+     S    interruptible sleep
+     T    stopped by job control signal
+     t    stopped by debugger during the tracing
+     X    dead (should never be seen)
+     Z    zombie
+    """
+    return task_state_to_char(task)
+
+
+def get_command(task: Object) -> str:
+    """
+    :returns: name of the command
+    """
+    return task.comm.string_().decode("utf-8")
+
+
+def show_ps_m_info(prog: Program) -> None:
```

Comment:
Assuming you're going to eventually expand this, so that it works as just a part of the "ps" command, then you probably want this to accept an iterable of task objects, and simply print them. Further, I don't think "ps_m" is at all a descriptive name, maybe `show_tasks_last_runtime()`?
```suggestion
def show_tasks_last_runtime(tasks: Iterable[Object]) -> None:
```

Then down below, rather than doing `for t in for_each_task(prog)`, you would do `for t in tasks`. This will allow you to implement the filtering logic separately (e.g. for user / kthread / group leader options `-k|-u|-g`).